server/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -309,6 +310,11 @@ public NodeEnvironment(Settings settings, Environment environment) throws IOExce
             if (DiscoveryNode.isMasterNode(settings) || DiscoveryNode.isDataNode(settings)) {
                 ensureAtomicMoveSupported(nodePaths);
             }
+
+            if (!DiscoveryNode.isDataNode(settings)) {
```

Comment:
we prefer the `X == false` convention over `!X` (see other code) in order to make the negation more visually prominent.